AdyenActions/Components/3DS2/Action handlers/3DS2+Delegated Authentication/ThreeDS2PlusDACoreActionHandler.swift

```diff
@@ -175,44 +290,49 @@
             let token: ThreeDS2Component.ChallengeToken
             do {
                 token = try Coder.decodeBase64(challengeAction.challengeToken) as ThreeDS2Component.ChallengeToken
+                
+                guard let sdkInput = token.delegatedAuthenticationSDKInput else {
+                    completionHandler(.success(challengeResult))
+                    return
+                }
+                if shouldShowRegistrationScreen {
+                    presenter.showRegistrationScreen(
+                        component: self,
+                        registerDelegatedAuthenticationHandler: { [weak self] in
+                            guard let self = self else { return }
+                            self.performDelegatedRegistration(sdkInput) { [weak self] result in
+                                self?.deliver(challengeResult: challengeResult,
+                                              delegatedAuthenticationSDKOutput: result.successResult,
+                                              deleteDelegatedAuthenticationCredentials: nil,
+                                              completionHandler: completionHandler)
+                            }
+                        },
+                        fallbackHandler: {
+                            completionHandler(.success(challengeResult))
+                        }
+                    )
+                } else if presenter.userInput == .deleteDA {
```

Comment:
Maybe also good to not compare against a specific case but have a computed property `presenter.userInput.shouldDeleteDelegatedAuthenticationCredentials` (or something shorter) with a switch to not have any new added user input to slip through accidentally

Reply:
Sure. Created a computed property for this. Thanks.